### PR TITLE
Fix Issue 9588 - format prints context pointer for struct

### PR DESCRIPTION
**Pull request by berni44**
Thanks to Simen Kjaeraas, who pointed out this solution - seems to be the most straight forward to me.

### CODE REVIEW

File: std/format.d [hunk 1]
```diff
@@ -4391,22 +4391,27 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
...
-                else
-                    put(w, separator~val.tupleof[i].stringof[4..$]);
-            }
+            static if (__traits(identifier, val.tupleof[i]) == "this")
+                continue;
             else
             {
-                static if (i+1 < val.tupleof.length && val.tupleof[i].offsetof == val.tupleof[i+1].offsetof)
-                    put(w, (i > 0 ? separator : "")~"#{overlap "~val.tupleof[i].stringof[4..$]);
+                static if (0 < i && val.tupleof[i-1].offsetof == val.tupleof[i].offsetof)
```
Comment (thewilsonator) [inline std/format.d]: Please make this an `else static if` rather than introduce an extra pair of braces.

File: std/format.d [hunk 2]
```diff
@@ -4391,22 +4391,27 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
...
+                        put(w, separator~val.tupleof[i].stringof[4..$]~"}");
+                    else
+                        put(w, separator~val.tupleof[i].stringof[4..$]);
+                }
                 else
                 {
-                    static if (i > 0)
-                        put(w, separator);
-                    formatElement(w, e, f);
+                    static if (i+1 < val.tupleof.length && val.tupleof[i].offsetof == val.tupleof[i+1].offsetof)
```
Comment (thewilsonator) [inline std/format.d]: Ditto

File: std/format.d [hunk 3]
```diff
@@ -4391,22 +4391,27 @@ if ((is(T == struct) || is(T == union)) && (hasToString!(T, Char) || !is(Builtin
...
+                        put(w, (i > 0 ? separator : "")~"#{overlap "~val.tupleof[i].stringof[4..$]);
+                    else
+                    {
+                        static if (i > 0)
```
Comment (thewilsonator) [inline std/format.d]: ditto

Comment (berni44) [reply std/format.d]: I think, this cannot be replaced, because of the following `formatElement` call.

Comment (thewilsonator) [reply std/format.d]: Oh I didn't see that

**Pull request merged**